app/src/components/features/rules/RuleBuilder/Header/ActionButtons/CreateRuleButton/actions/index.js

```diff
@@ -52,7 +52,21 @@ export const transformAndValidateRuleFields = async (ruleData) => {
           });
       };
 
+      const getIndentedCode = (code) => {
+        if (!code || typeof code !== "string") return "";
+        code.trim();
+        // indenting the code without adding unnecessary tabs and new lines
+        const newCode = code
+          .replace(/^\n+|\n+$/g, "")
+          .split("\n")
+          .map((line) => (line.startsWith("\t") ? line : "\t" + line))
+          .join("\n");
+
+        return newCode;
+      };
```

Comment:
tried this, but this isn't working as intended, since I just want **extra** indentation to the code that will be injected inside the HTML tags, all while not interferring with the way the user initially intended his code to be formatted.

I know this adds a little bit of messy code with all the regex and string concatenation, but this is meant to (consistently) fix preview indentation without changing the look of the code entirely